src/auth.js

```diff
@@ -15,46 +15,70 @@
+    const creds = await getApplicationDefaultCredentials();
+    console.log(`Retrieved application default credentials.`)
+    return creds;
   } catch (err) {
-    throw new Error(
-        'Fail to get credentials. Please run: \n' +
-        '`gcloud auth application-default login` or \n' +
-        '`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`');
+    console.log(`Failed to retrieve application default credentials: ${err.message}.`);
   }
-
-  let creds;
   try {
-    creds = (await client.getAccessToken()).token;
+    console.log(`Retrieving credentials from gcloud...`)
+    const creds = await getGcloudCredentials();
+    console.log(`Retrived credentials from the current active account logged into gcloud.`)
+    return creds;
   } catch (err) {
+    console.log(`Failed to retrieve credentials from gcloud: ${err.message}.`)
+  }
+  throw new Error(
+      'Fail to get credentials. Please run: \n' +
+      '`gcloud auth application-default login` or \n' +
+      '`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`');
+}
+
+/**
+ * Retrieves the application default credentials.
+ *
+ * @return {!Promise<string>} cred Encrypted access token.
+ */
+async function getApplicationDefaultCredentials() {
+  const auth = new GoogleAuth({
+    scopes: 'https://www.googleapis.com/auth/cloud-platform',
+    projectId: 'unused-project'
+  });
+  const client = await auth.getClient();
+  const creds = (await client.getAccessToken()).token;
+  const tokenScopes = (await client.getTokenInfo(creds)).scopes;
+  if (!tokenScopes.includes(
+          'https://www.googleapis.com/auth/cloud-platform')) {
     throw new Error(
-        'Fail to get credentials. Please run: \n' +
-        '`gcloud auth application-default login` or \n' +
-        '`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`');
-  } finally {
-    const tokenScopes = (await client.getTokenInfo(creds)).scopes;
-    if (!tokenScopes.includes(
-            'https://www.googleapis.com/auth/cloud-platform')) {
-      throw new Error(
-          'Token has insufficient authentication scopes.\n' +
-          'Please configure access scope following instructions on ' +
-          'https://cloud.google.com/artifact-registry/docs/access-control#compute');
-    }
+        'Token has insufficient authentication scopes.\n' +
+        'Please configure access scope following instructions on ' +
+        'https://cloud.google.com/artifact-registry/docs/access-control#compute');
   }
-  return creds;
+  return creds; 
+}
+
+/**
+ * Retrieves the credentials from the current active account logged into gcloud.
+ *
+ * @return {!Promise<string>} cred Encrypted access token.
+ */
+async function getGcloudCredentials() {
+  const isWindows = process.platform === 'win32';
+  let gcloud = isWindows? "gcloud.cmd" : "gcloud";
+  const {stdout, stderr} = await exec(`${gcloud} auth print-access-token`);
+  return stdout;
```

Comment:
Would this function throw any error in case exec fails? If not, when would the above catch err gets reached? `Failed to retrieve credentials from gcloud: ${err.message}.`

Reply:
Yes, if `gcloud auth login` has not been called before. It could also be possible that gcloud was never installed on user's machine.